### PR TITLE
[IA-1973] Fix jupyter extension installs

### PR DESCRIPTION
**Pull request by gpcarr**
Now that we docker compose with env var `PIP_TARGET` we need to update the `gce-init` script to use `PIP_TARGET` instead of `PIP_USER` when installing packages

Tested the following:
- `NotebookGCECustomizationSpec` now passes
- `NotebookPyKernelSpec` still passes
- determined `ROOT_USER_PIP_DIR` and `JUPYTER_USER_PIP_DIR` by reverting my FIAB to before I updated user package installation persistence and seeing where `userJupyterExtensionConfig` get installed to

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by Qi77Qi** [comment]
does https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/resources/init-resources/init-actions.sh need the same change?

**Comment by gpcarr** [comment]
> does https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/resources/init-resources/init-actions.sh need the same change?

is `init-actions.sh` only used for GCE VMs? what about `startup.sh`?

**Comment by gpcarr** [comment]
> does https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/resources/init-resources/init-actions.sh need the same change?

seems like `init-actions.sh` is only for dataproc clusters https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/resources/init-resources/init-actions.sh#L6

**Comment by gpcarr** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
@gpcarr yea, gce-init.sh is for gce vms, and init-actions.sh is for dataproc clusters, they both use the same images. regarding `startup.sh`, if we're installing Jupiter extension there, then we probably should also update it

**Comment by gpcarr** [comment]
jenkins retest

**Comment by gpcarr** [comment]
> @gpcarr yea, gce-init.sh is for gce vms, and init-actions.sh is for dataproc clusters, they both use the same images. regarding `startup.sh`, if we're installing Jupiter extension there, then we probably should also update it

the changes that broke the gce vms were not in the images but in `jupyter-docker-compose-gce` where I set `PIP_USER` to false and set a `PIP_TARGET`, all of the images in `terra-docker` still have `ENV PIP_USER=true` at the end of them

**Comment by gpcarr** [comment]
this comment in `startup.sh` makes it seem like it's only for dataproc clusters but i'm not sure that's true https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/resources/init-resources/startup.sh#L6 @rtitle does `startup.sh` also run on gce vms? if so, it does some installations for jupyter and i'll need to update it as well along with passing a template param telling us if it's running on gce or dataproc because the change is only for gce

**Comment by gpcarr** [comment]
fiab-start failure, jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookBioconductorKernelSpec.scala [hunk 1]
```diff
@@ -108,19 +108,19 @@ class NotebookBioconductorKernelSpec extends RuntimeFixtureSpec with NotebookTes
             notebookPage.executeCell("library('BiocSklearn')")
 
             notebookPage.executeCell("""BiocManager::install('rhdf5')""", installTimeout)
-            notebookPage.executeCell("library(rhdf5)")
+            notebookPage.executeCell("library('rhdf5')")
```
Comment (Qi77Qi) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookBioconductorKernelSpec.scala]: @nturaga just curious, is `'` needed here?

Comment (gpcarr) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookBioconductorKernelSpec.scala]: it shouldn't be https://stackoverflow.com/questions/25210455/how-can-library-accept-both-quoted-and-unquoted-strings

**Comment by Qi77Qi** [comment]
just occurred to me, startup script is shared between dataproc and gce VM, so not sure how you would differentiate the two

**Comment by gpcarr** [comment]
> just occurred to me, startup script is shared between dataproc and gce VM, so not sure how you would differentiate the two

just pushed a change to account for differentiating the two. testing it now

File: http/src/main/resources/init-resources/jupyter-docker-compose-gce.yaml [hunk 2]
```diff
@@ -28,6 +28,7 @@ services:
       PIP_USER: "false"
       PIP_TARGET: "${NOTEBOOKS_DIR}/packages"
       R_LIBS: "${NOTEBOOKS_DIR}/packages"
+      PYTHONPATH: "${PYTHONPATH:+$PYTHONPATH:}${NOTEBOOKS_DIR}/packages"
```
Comment (rtitle) [inline http/src/main/resources/init-resources/jupyter-docker-compose-gce.yaml]: Trying to understand this syntax.. does this result in a `PYTHONPATH` of `/etc/jupyter/custom:/home/jupyter-user/notebooks` in the container?

**Review by rtitle — APPROVED** [review]
This looks good to me

File: http/src/main/resources/init-resources/jupyter-docker-compose-gce.yaml (hunk 2 above)
Comment (gpcarr) [reply http/src/main/resources/init-resources/jupyter-docker-compose-gce.yaml]: yeah, i added this in this PR so we can account for users that use old docker images that don't have the `/packages` dir in their `PYTHONPATH`

Comment (rtitle) [reply http/src/main/resources/init-resources/jupyter-docker-compose-gce.yaml]: Yeah makes sense to add it here, was just confused about the bash syntax

Comment (gpcarr) [reply http/src/main/resources/init-resources/jupyter-docker-compose-gce.yaml]: actually removed this line and punting on it for now. it was breaking gce vm creation so need to figure out how to do it correctly but for now going to get the original fix in

**Pull request merged**